plugins/hydra_orion_sweeper/hydra_plugins/hydra_orion_sweeper/_impl.py

```diff
@@ -0,0 +1,335 @@
+    ReservationRaceCondition,
+    WaitingForTrials,
+    BrokenExperiment,
+)
+
+from .config import OrionClientConf, WorkerConf, AlgorithmConf, StorageConf
+
+log = logging.getLogger(__name__)
+
+
+def as_overrides(trial, additional):
+    """Returns the trial arguments as hydra overrides"""
+    kwargs = deepcopy(additional)
+    kwargs.update(flatten(trial.params))
+    return tuple(f"++{k}={v}" for k, v in kwargs.items())
+
+
+class SpaceParser:
+    """Generate an Orion space from parameters and overrides"""
+
+    def __init__(self) -> None:
+        self.base_space = dict()
+        self.overrides = dict()
+        self.arguments = dict()
+
+    def space(self) -> Space:
+        """Generate the final space after overrides that will be used for the optimization"""
+        configuration = deepcopy(self.base_space)
+        log.info("Orion base space is %s", configuration)
+        configuration.update(self.overrides)
+        log.info("Orion space overrides are %s", self.overrides)
+        return SpaceBuilder().build(configuration), self.arguments
+
+    def add_from_parametrization(self, parametrization: Optional[DictConfig]) -> None:
+        """Use the parametrization retrieved from the configuration to generate a
+        preliminary research space
+
+        """
+        for k, v in parametrization.items():
+            try:
+                dim = DimensionBuilder().build(k, v)
+                self.base_space[dim.name] = dim.get_prior_string()
+            except (TypeError, NameError):
+                # Regular argument
+                self.arguments[k] = v
+
+    def add_from_overrides(self, arguments: List[str]) -> None:
+        """Create a dictionary of overrides to modify the research space"""
+        parser = OverridesParser.create()
+        parsed = parser.parse_overrides(arguments)
+
+        for override in parsed:
+            dim = self.process_overrides(override)
+
+            if dim is None:
+                self.arguments[override.get_key_element()] = override.value()
+            else:
+                self.overrides[dim.name] = dim.get_prior_string()
+
+    def process_overrides(self, override: Override) -> Dimension:
+        """Identify the sweep overrides and build a matching dimension"""
+        values = override.value()
+        name = override.get_key_element()
+
+        def build_dim(name):
+            builder = DimensionBuilder()
+            builder.name = name
+            return builder
+
+        if override.is_choice_sweep():
+            return build_dim(name).choices(*values.list)
+
+        elif override.is_range_sweep():
+            choices = [v for v in range(values.start, values.stop, values.step)]
+            return build_dim(name).choices(*choices)
+
+        elif override.is_interval_sweep():
+            discrete = type(values.start) is int
+            log = "log" in values.tags
+
+            cast_type = float
+            if discrete or values.start % 1 == values.end % 1 == 0.0:
```

Comment:
`interval(1.0, 2.0)` would be casted to int?

Reply:
Yes, but that does not matter since if discrete==False the result sampled will still be floats
This is just a cosmetic cast so ``space.configuration == input``